src/audio/sr_audio/sr_audio.c

```diff
@@ -0,0 +1,720 @@
+void SR_HIFI2_AsyncClear(SR_HIFI2_Context_t* libCtx, int flags);
+
+// The below two functions allow sending/receiving parameter data in fragments.
+//
+// paramId: parameter id
+// paramData: points to parameter data (or to current data fragment, if data is fragmented)
+// dataSize: overall data size
+// dataFragmentSize: size of current fragment
+// isFirstFragment: 1 if paramData points to first fragment, 0 otherwise
+//
+// If non-fragmented data (aka single fragment) is passed, just call functions as follows below:
+// SR_HIFI2_SetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//   or
+// SR_HIFI2_GetParameter(context, paramId, paramData, dataSize, dataSize, 1);
+//
+void SR_HIFI2_SetParameter(SR_HIFI2_Context_t* libCtx, int paramId, const uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
+void SR_HIFI2_GetParameter(SR_HIFI2_Context_t* libCtx, int paramId, uint8_t* paramData, int dataSize, int dataFragmentSize, int isFirstFragment);
+
+// Toggles bypass mode (bypass = 1 for bypassing, 0 otherwise)
+void SR_HIFI2_Bypass(SR_HIFI2_Context_t* libCtx, int bypass);
+
+// Processing function.
+// frameCount: number of samples in each input/output buffers
+void SR_HIFI2_Process(SR_HIFI2_Context_t* libCtx, int frameCount, const int32_t* inputL, const int32_t* inputR, int32_t* outputL, int32_t* outputR);
+
+uint32_t xtensa_get_ccount();
+
+#ifdef _ADD_TEST_CLICK_AT48
+static void _add_test_click_at47(int32_t* samplesL, int32_t* samplesR) {
+  samplesL[47] = 0;
+  samplesR[47] = 0;
+}
+#endif
+
+#define maxChannelCount 2
+#define bufferFrameCount 48
+
+// SRAUDIO component private data
+struct comp_data {
+  struct comp_data_blob_handler* model_handler;
+  struct sof_sraudio_config* config;
+  size_t configSize;
+  enum sof_ipc_frame source_format; // source frame format
+  enum sof_ipc_frame sink_format;   // sink frame format
+  SR_HIFI2_Context_t sraudio_ctx;   // processing context
+  int32_t inProcessBuffer[maxChannelCount][bufferFrameCount];
+  int32_t outProcessBuffer[maxChannelCount][bufferFrameCount];
+  sraudio_func sraudio_func;        // processing function
+};
+
+#if CONFIG_FORMAT_S16LE
+static void sraudio_s16_default(const struct comp_dev* dev, const struct audio_stream* source, struct audio_stream* sink, uint32_t frameCount) {
+  struct comp_data* cd = comp_get_drvdata(dev);
+  int32_t ch, i, j;
+  uint32_t inChannelCount = source->channels;
+  uint32_t outChannelCount = sink->channels;
+  uint32_t inBufferSampleCount = bufferFrameCount * inChannelCount;
+  uint32_t outBufferSampleCount = bufferFrameCount * outChannelCount;
+  uint32_t tailFrameCount = frameCount;
+  uint32_t inOffset = 0;
+  uint32_t outOffset = 0;
+  SR_HIFI2_Context_t* ctx = &cd->sraudio_ctx;
+  for (; tailFrameCount >= bufferFrameCount; tailFrameCount -= bufferFrameCount, inOffset += inBufferSampleCount, outOffset += outBufferSampleCount) {
```

Comment:
We need to split this function into logical blocks separated by newlines and a comment. This way it's easier to follow and review. It also looks like there is some buffer copying here ?

Reply:
That's basically splitting a big buffer into smaller 48-sample-frame buffers. The names of variables are pretty self explanatory, such as bufferFrameCount, tailFrameCount, etc. We prefer an easily readable code once you know what it does instead of cluttering it with comments. Sort of reading the original (code) instead of its translation (comments). Will add the comments if that's the requirement, thou.

Reply:
thanks, this is to help my brains pattern matching when reviewing. I find it difficult when it's all one big block with no comments..